### PR TITLE
[DF-619] Added Related Resources Column

### LINKED ISSUE #619: [DF-421] add a "related resources" column to the UI's events table
This column in the events table component will contain links to resources for `InternalEventType` events. For example `BadWorkspaceConfiguration` events for `MessageTemplateRenderError` should contain a `templateId` in its payload. This id should be used to render a link to the related template with its name as the link title.

Moreover the "individual event view" rendered by clicking on an event, should also contain a related resources section.

Note that this should also relate journeys by `journeyId`. Journey resources should link back to a broadcast page rather than a journey page, if the journey belongs to a broadcast.

<sub>From [SyncLinear.com](https://synclinear.com) | [DF-421](https://linear.app/dittofeed/issue/DF-421/add-a-related-resources-column-to-the-uis-events-table)</sub>

### PR DESCRIPTION
**Pull request by mehulmathur16**
/claim #619
/fixes #619

https://github.com/dittofeed/dittofeed/assets/64700961/98954be4-2f2d-48f7-bd7f-ac3feb3b2302

### CODE REVIEW

File: packages/dashboard/src/pages/events.page.tsx [hunk 1]
```diff
@@ -7,16 +9,30 @@ import { EventsTable } from "../components/eventsTable";
...
 
 export const getServerSideProps: GetServerSideProps<PropsWithInitialState> =
-  requestContext(async (_ctx, dfContext) => ({
-    props: addInitialStateToProps({
-      dfContext,
-      props: {},
-      serverInitialState: {},
-    }),
-  }));
+  requestContext(async (_ctx, dfContext) => {
```
Comment (maxgurewitz) [inline packages/dashboard/src/pages/events.page.tsx:15]: Can you also be sure to make these changes to the `/dashboard/users/[id]` page, which also displays events?

File: packages/dashboard/src/components/eventsTable.tsx [hunk 2]
```diff
@@ -159,6 +166,34 @@ export function EventsTable({
       field: "messageId",
       flex: 1,
     },
+    {
+      field: "relatedResources",
+      flex: 1,
+      valueGetter: (params: any) => JSON.parse(params.row.traits),
+      renderCell: ({ value }: GridRenderCellParams) => {
```
Comment (maxgurewitz) [inline packages/dashboard/src/components/eventsTable.tsx:173]: I realize that I should have been more explicit in the issue, but also need to relate segment and journey/broadcast resources as now described in the ticket.

**Review by maxgurewitz — CHANGES_REQUESTED** [review]
Can you post a recording showing a `BadWorkspaceConfiguration` event in the table specifically? You can trigger this by using a template that references a user property which is not defined on the user. Thank you!

**Review by maxgurewitz — CHANGES_REQUESTED** [review]
We also need to add the related resources to the sidebar.

**Comment by mehulmathur16** [comment]
> Can you post a recording showing a `BadWorkspaceConfiguration` event in the table specifically? You can trigger this by using a template that references a user property which is not defined on the user. Thank you!


https://github.com/dittofeed/dittofeed/assets/64700961/14f0a0f4-418b-49ec-b457-9a3eb7041ff1

**Review by maxgurewitz — APPROVED** [review]

**Pull request merged**